enclave/include/enclave_context.h

```diff
@@ -123,10 +123,11 @@ class EnclaveContext {
     //}
 
     bool get_client_key(uint8_t* key) {
-      if (client_key_is_set)
+      if (client_key_is_set) {
           memcpy(key, client_key, CIPHER_KEY_SIZE);
-      else
-          memset(key, 0, CIPHER_KEY_SIZE);
+          return true;
+      }
+      return false;
```

Comment:
Should we `LOG(FATAL)` or something here to tell the user there's no key for that client?

Reply:
I thought about that, but this might be preferable because it's not really a "fatal" error -- in the sense that the user can easily recover from it by simply adding the key.

Reply:
If the enclave cannot find the key, all subsequent steps will fail because the DMatrix can't be loaded. What will happen if the file can't be decrypted properly? Will the execution terminate with a `mbed_tls_gcm_decrypt` error? If so, does user recovery simply mean the user has to start another process? I think we should somehow more gracefully exit here, perhaps with some kind of logging statement.

Reply:
The calling function should check the return value, and if it is `false`, then it should handle the error accordingly.  For example, the calling functions in `enclave/c_api.cc` return a `-1` if this function returns false.

Reply:
Should we leave this up to the user? Nothing can really be done if the file can't be decrypted though right?

Reply:
Right, but leaving it up to the user means that the application should be able to decide whether to re-invoke the API if the key is not found, or to kill the process. No?

Reply:
I suppose doing it this way offers the user more flexibility. We can keep it like this!

Reply:
Actually, you were right. I added a `LOG(FATAL)` to the `get_client_key` function. The `API_END` macro catches the error and returns a -1 to the application code (which is what I wanted), instead of terminating the program :P